core/src/main/java/com/graphhopper/storage/index/LocationIndexTree.java

```diff
@@ -564,6 +572,133 @@ protected boolean check(int node, double normedDist, int wayIndex, EdgeIteratorS
         return closestMatch;
     }
 
+    /**
+     * Returns all edges that are within the specified radius around the queried position.
+     *
+     * @param queryLat latitude to search from
+     * @param queryLon longitude to search from
+     * @param edgeFilter only include edges from edgeFilter
+     * @param radius include all results within this radius (in meters) - assuming it's not
+     *        bounded by maxIterations.
+     * @param maxIterations if not null, search only a square of (2 * maxIterations - 1) tiles
```

Comment:
> I don't get the purpose of maxIterations and radius.

The purpose is to find everything within `radius`. The `maxIterations` is there to bound to a certain amounts of tiles - as discussed in MapMatching 65. (Personally, I don't think it's required - as long as we always search one tile index 'greater' than the radius - but that's what was in the code - and I'd prefer to keep this a moving of the method as opposed to a rewrite. Yes, I did rewrite it somewhat so that the method actually reflected it's purpose and implementation.)